### PR TITLE
[dataset/cleanup] Remove usage from dataset impl

### PR DESCRIPTION
**Pull request by zhoonit**
- [dataset/cleanup] Remove usage from dataset impl

```
As each dataset now has one usage(train, valid or test).
This patch removes usage handles from dataset impl.

**Major Changes**
1. merge train/test/val members to one
2. clarify some variable names.
3. move global variables to members of an instance.
4. remove databuffer_utils.h as it is no longer used.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

File: nntrainer/dataset/databuffer.h [hunk 1]
```diff
@@ -201,101 +184,48 @@ class DataBuffer : public ml::train::Dataset {
...
-  std::thread test_thread;
+  /**< @todo below variable should be owned by databuffer that has fixed size */
+  unsigned int samples_per_epoch; /**< size of samples in the dataset */
+  unsigned int remaining_samples_per_epoch; /**< size of samples remaining in
+                                               current epoch */
 
-  std::vector<std::string> labels;
-  bool validation[NBUFTYPE];
+  bool is_running; /**< flag to check if running */
+  bool is_valid;   /**< check if current buffer is in valid state */
```
Comment (kparichay) [inline nntrainer/dataset/databuffer.h:228]: how about is_valid -> initialized?

Comment (zhoonit) [reply nntrainer/dataset/databuffer.h:228]: done. fyi) in the upcoming design this will not be used :)

Comment (kparichay) [reply nntrainer/dataset/databuffer.h]: yes, that will be even better.

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM with small suggestions.

**Comment by zhoonit** [comment]
> Great. one small suggestion to have more unit test cases with this update. ( Cause we had issues related this dataset )

Thanks, as this is on the refactoring progress, I'll add test while refactoring :)

**Pull request merged**